bundles/org.openhab.binding.speedtest/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,158 @@
+		<state readOnly="true" pattern="%.2f ms"></state>
+	</channel-type>
+	<channel-type id="downloadBandwidth">
+		<item-type>Number:DataTransferRate</item-type>
+		<label>Download Bandwidth</label>
+		<description>Download bandwidth, e.g. in Mbit/s</description>
+		<state readOnly="true" pattern="%.2f Mbit/s"></state>
+	</channel-type>
+	<channel-type id="downloadBytes" advanced="true">
+		<item-type>Number:DataAmount</item-type>
+		<label>Downloaded Bytes</label>
+		<description>Amount of data that was used for the last upload bandwidth test</description>
+		<state readOnly="true" pattern="%.0f MB"></state>
+	</channel-type>
+	<channel-type id="downloadElapsed" advanced="true">
+		<item-type>Number:Time</item-type>
+		<label>Download Elapsed Time</label>
+		<description>Time spent for the last download bandwidth test</description>
+		<state readOnly="true" pattern="%.2f s"></state>
+	</channel-type>
+	<channel-type id="uploadBandwidth">
+		<item-type>Number:DataTransferRate</item-type>
+		<label>Upload Bandwidth</label>
+		<description>Upload bandwidth, e.g. in Mbit/s</description>
+		<state readOnly="true" pattern="%.2f Mbit/s"></state>
+	</channel-type>
+	<channel-type id="uploadBytes" advanced="true">
+		<item-type>Number:DataAmount</item-type>
+		<label>Uploaded Bytes</label>
+		<description>Amount of data that was used for the last upload bandwidth test</description>
+		<state readOnly="true" pattern="%.0f MB"></state>
+	</channel-type>
+	<channel-type id="uploadElapsed" advanced="true">
+		<item-type>Number:Time</item-type>
+		<label>Upload Elapsed Time</label>
+		<description>Time spent for the last upload bandwidth test</description>
+		<state readOnly="true" pattern="%.2f s"></state>
+	</channel-type>
+	<channel-type id="isp">
+		<item-type>String</item-type>
+		<label>ISP</label>
+		<description>Your Internet Service Provider (ISP) as calculated by Ookla</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="interfaceInternalIp" advanced="true">
+		<item-type>String</item-type>
+		<label>Interface Internal IP</label>
+		<description>IP address of the internal interface that was used for the test</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="interfaceExternalIp" advanced="true">
+		<item-type>String</item-type>
+		<label>Interface External IP</label>
+		<description>IP address of the external interface that was used for the test</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="resultUrl">
+		<item-type>String</item-type>
+		<label>Result URL</label>
+		<description>The URL to the Speedtest results in HTML on the Ookla webserver</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="triggerTest" advanced="true">
+		<item-type>Switch</item-type>
+		<label>Trigger Test</label>
+		<description>Trigger in order to run Speedtest manually</description>
```

Comment:
This seems like a stateless channel? In this case you could apply this policy:
```suggestion
		<description>Trigger in order to run Speedtest manually</description>
		<autoUpdatePolicy>veto</autoUpdatePolicy>
```

See https://www.openhab.org/docs/developer/bindings/thing-xml.html#auto-update-policies

Reply:
`veto` would be one possiblity to keep the state to `OFF` even when receiving the `ON` command.

Right now it is implemented in a way, that the framework sets the state to `ON` and the binding sets the sate back to `OFF` after the test was executed:
```
        if (ch.equals(SpeedtestBindingConstants.TRIGGER_TEST)) {
            if (command instanceof OnOffType) {
                if (command == OnOffType.ON) {
                    getSpeed();
                    updateState(channelUID, OnOffType.OFF);
                }
            }
        }
```

Is there a preferred solution?

Reply:
Not that I'm aware of, I think both a valid solutions, and in both cases I believe you need to handle concurrent state changes (i.e. receiving ON while a speed test is already in progress).